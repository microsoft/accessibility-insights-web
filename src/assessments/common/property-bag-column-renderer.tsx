// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { ColumnValueBag } from 'common/types/property-bag/column-value-bag';
import { PropertyBagColumnRendererConfig } from 'common/types/property-bag/property-bag-column-renderer-config';
import { isEmpty } from 'lodash';
import * as React from 'react';

import { DictionaryStringTo } from 'types/common-types';

export function propertyBagColumnRenderer<TPropertyBag extends ColumnValueBag>(
    item: InstanceTableRow<TPropertyBag>,
    configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
): JSX.Element {
    const mapper = (config: PropertyBagColumnRendererConfig<TPropertyBag>, index: number) => {
        if (item.instance.propertyBag == null) {
            return null;
        }
        const value = item.instance.propertyBag[config.propertyName];
        if (value == null && config.defaultValue == null) {
            return null;
        }

        return render(config, value, index);
    };

    const render = (
        config: PropertyBagColumnRendererConfig<TPropertyBag>,
        value: any,
        index: number,
    ) => {
        return (
            <div key={`property-${index}`} className="property-bag-div">
                <span className="display-name">{`${config.displayName}: `}</span>
                {renderValue(config, value)}
            </div>
        );
    };

    const renderValue = (config: PropertyBagColumnRendererConfig<TPropertyBag>, value: any) => {
        if (config.expand) {
            return renderProperties(config, value);
        }

        return <React.Fragment>{value || config.defaultValue}</React.Fragment>;
    };

    const renderProperties = (
        config: PropertyBagColumnRendererConfig<TPropertyBag>,
        propertyMap: DictionaryStringTo<string>,
    ) => {
        if (isEmpty(propertyMap)) {
            return <React.Fragment>{config.defaultValue.toString()}</React.Fragment>;
        }

        return Object.keys(propertyMap).map(key => {
            return (
                <div key={key} className="expanded-property-div">
                    {renderInnerKeyValue(key, propertyMap[key])}
                </div>
            );
        });
    };

    const renderInnerKeyValue = (key: string, value: string) => {
        if (value === null) {
            return <span className="display-name">{`${key}`}</span>;
        } else {
            return (
                <React.Fragment>
                    <span className="display-name">{`${key}: `}</span>
                    {value}
                </React.Fragment>
            );
        }
    };

    return <div className="property-bag-container">{configs.map(mapper)}</div>;
}
