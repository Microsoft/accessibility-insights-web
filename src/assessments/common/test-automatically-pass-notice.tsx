// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';
import * as Markup from '../markup';

const TestAutomaticallyPassedNotice = (
    <p>
        <Markup.Emphasis>
            Note: If no matching/failing instances are found, this requirement will automatically be marked as pass.
        </Markup.Emphasis>
    </p>
);

export default TestAutomaticallyPassedNotice;
