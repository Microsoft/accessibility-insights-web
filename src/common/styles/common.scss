// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import 'high-contrast-override.scss';
@import 'colors.scss';
@import 'fonts.scss';

$focus-ring-color: $neutral-100;

@mixin set-focus-ring-styles($property: 'border') {
    #{$property}: 1px solid $focus-ring-color;
    @include ms-high-contrast-override {
        #{$property}: 1px dashed windowtext !important;
    }
}

.ms-Fabric.is-focusVisible .ms-Nav-group .ms-nav-linkButton:focus::after {
    @include set-focus-ring-styles('border');
}

button::after {
    content: '';
    position: absolute;
}

.ms-Button.ms-Button--primary:not(.is-disabled) {
    background: $communication-primary;

    &:hover {
        background: $communication-shade-10;
    }
}

.ms-fontColor-neutralPrimary,
.ms-fontColor-neutralPrimary--hover:hover {
    color: $neutral-100 !important;
}

.ms-Fabric {
    color: $neutral-100;
}

.generic-toggle-component {
    margin-bottom: 4vh;
    .toggle-container {
        margin-bottom: 1vh;
        .toggle-name {
            display: inline-block;
            font-size: 17px;
            font-weight: bold;
        }

        .toggle {
            float: right;
        }
    }
}

.high-contrast-theme {
    .insights-link {
        &:hover {
            text-decoration: underline;
        }
    }
    .is-selected {
        .status-icon {
            color: $high-contrast-icon-color !important;
        }
    }
}

.insights-link {
    // marking important to override office fabric style
    color: $link !important;
    text-decoration: none;

    &:hover {
        color: $link-hover !important;
        text-decoration: underline;
    }
}

.ms-Spinner {
    .ms-Spinner-circle {
        border-top-color: $spinner-text;
    }

    .ms-Spinner-label {
        color: $spinner-text;
    }
}

$normalTitleFontWeight: 300 !default;

$pivotItemLeftBorderWidth: 3px !default;
$pivotItemBorderStyle: solid !default;

$detailsViewHeaderBarHeight: 40px !default;
$detailsViewNavPivotsHeight: 41px !default;
$detailsViewCommandBarHeight: 40px !default;
$detailsViewCommandBarBorderHeight: 1px !default;
$detailsViewLeftNavWidth: 200px !default;
$detailsViewTotalHeaderHeight: (#{$detailsViewCommandBarBorderHeight} + #{$detailsViewCommandBarHeight} + #{$detailsViewHeaderBarHeight});

$fastPassRightPanelMarginRight: 24px !default;
$fastPassRightPanelMarginLeft: 24px !default;
$fastPassRightPanelMarginTop: 24px !default;

@mixin ellipsedText {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}
