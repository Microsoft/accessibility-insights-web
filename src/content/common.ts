// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as react from 'react';

import { link } from './link';
import { ContentCreator } from '../views/content/content-page';
import { GuidanceTitle as gt } from '../views/content/guidance-title';

export const create = ContentCreator(link);
export const React = react;
export const GuidanceTitle = gt;
