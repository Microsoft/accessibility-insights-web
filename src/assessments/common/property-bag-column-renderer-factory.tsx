// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ColumnValueBag } from '../../common/types/property-bag/column-value-bag';
import { IAssessmentInstanceRowData } from '../../DetailsView/components/assessment-instance-table';
import { propertyBagColumnRenderer, PropertyBagColumnRendererConfig } from './property-bag-column-renderer';

export class PropertyBagColumnRendererFactory {
    public static get<TPropertyBag extends ColumnValueBag>(
        configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
    ): (item: IAssessmentInstanceRowData<TPropertyBag>) => JSX.Element {
        return item => {
            return propertyBagColumnRenderer(item, configs);
        };
    }
}
