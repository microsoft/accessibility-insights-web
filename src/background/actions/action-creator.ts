// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { CardSelectionActions } from 'background/actions/card-selection-actions';
import { SidePanelActions } from 'background/actions/side-panel-actions';
import { UnifiedScanResultActions } from 'background/actions/unified-scan-result-actions';
import { TestMode } from 'common/configs/test-mode';
import { VisualizationConfigurationFactory } from 'common/configs/visualization-configuration-factory';
import * as TelemetryEvents from 'common/extension-telemetry-events';
import { Logger } from 'common/logging/logger';
import { getStoreStateMessage, Messages } from 'common/messages';
import { NotificationCreator } from 'common/notification-creator';
import { StoreNames } from 'common/stores/store-names';
import { VisualizationType } from 'common/types/visualization-type';
import { ScanCompletedPayload } from 'injected/analyzers/analyzer';
import { DictionaryNumberTo } from 'types/common-types';
import { VisualizationActions } from '../actions/visualization-actions';
import { VisualizationScanResultActions } from '../actions/visualization-scan-result-actions';
import { ExtensionDetailsViewController } from '../extension-details-view-controller';
import { Interpreter } from '../interpreter';
import { TargetTabController } from '../target-tab-controller';
import { TelemetryEventHandler } from '../telemetry/telemetry-event-handler';
import { ActionHub } from './action-hub';
import {
    AddTabbedElementPayload,
    BaseActionPayload,
    OnDetailsViewOpenPayload,
    OnDetailsViewPivotSelected,
    RescanVisualizationPayload,
    ToggleActionPayload,
    VisualizationTogglePayload,
} from './action-payloads';
import { InspectActions } from './inspect-actions';

const visualizationMessages = Messages.Visualizations;

export class ActionCreator {
    private visualizationActions: VisualizationActions;
    private visualizationScanResultActions: VisualizationScanResultActions;
    private adHocTestTypeToTelemetryEvent: DictionaryNumberTo<string> = {
        [VisualizationType.Color]: TelemetryEvents.COLOR_TOGGLE,
        [VisualizationType.Headings]: TelemetryEvents.HEADINGS_TOGGLE,
        [VisualizationType.Issues]: TelemetryEvents.AUTOMATED_CHECKS_TOGGLE,
        [VisualizationType.Landmarks]: TelemetryEvents.LANDMARKS_TOGGLE,
        [VisualizationType.TabStops]: TelemetryEvents.TABSTOPS_TOGGLE,
        [VisualizationType.NeedsReview]: TelemetryEvents.NEEDS_REVIEW_TOGGLE,
    };
    private inspectActions: InspectActions;
    private cardSelectionActions: CardSelectionActions;
    private unifiedScanResultActions: UnifiedScanResultActions;
    private sidePanelActions: SidePanelActions;

    constructor(
        private readonly interpreter: Interpreter,
        readonly actionHub: ActionHub,
        private readonly detailsViewController: ExtensionDetailsViewController,
        private readonly telemetryEventHandler: TelemetryEventHandler,
        private readonly notificationCreator: NotificationCreator,
        private readonly visualizationConfigurationFactory: VisualizationConfigurationFactory,
        private readonly targetTabController: TargetTabController,
        private readonly logger: Logger,
    ) {
        this.visualizationActions = actionHub.visualizationActions;
        this.visualizationScanResultActions = actionHub.visualizationScanResultActions;
        this.inspectActions = actionHub.inspectActions;
        this.cardSelectionActions = actionHub.cardSelectionActions;
        this.unifiedScanResultActions = actionHub.scanResultActions;
        this.sidePanelActions = actionHub.sidePanelActions;
    }

    public registerCallbacks(): void {
        this.interpreter.registerTypeToPayloadCallback(
            visualizationMessages.Common.Toggle,
            this.onVisualizationToggle,
        );
        this.interpreter.registerTypeToPayloadCallback(
            visualizationMessages.Common.ScanCompleted,
            this.onAdHocScanCompleted,
        );
        this.interpreter.registerTypeToPayloadCallback(
            visualizationMessages.Common.ScrollRequested,
            this.onScrollRequested,
        );
        this.interpreter.registerTypeToPayloadCallback(
            visualizationMessages.Common.RescanVisualization,
            this.onRescanVisualization,
        );

        this.interpreter.registerTypeToPayloadCallback(
            visualizationMessages.Issues.UpdateFocusedInstance,
            this.onUpdateFocusedInstance,
        );

        this.interpreter.registerTypeToPayloadCallback(
            getStoreStateMessage(StoreNames.VisualizationStore),
            this.getVisualizationToggleCurrentState,
        );
        this.interpreter.registerTypeToPayloadCallback(
            getStoreStateMessage(StoreNames.VisualizationScanResultStore),
            this.getScanResultsCurrentState,
        );

        this.interpreter.registerTypeToPayloadCallback(
            visualizationMessages.TabStops.TabbedElementAdded,
            this.onTabbedElementAdded,
        );
        this.interpreter.registerTypeToPayloadCallback(
            visualizationMessages.TabStops.RecordingCompleted,
            this.onRecordingCompleted,
        );
        this.interpreter.registerTypeToPayloadCallback(
            visualizationMessages.TabStops.TerminateScan,
            this.onRecordingTerminated,
        );

        this.interpreter.registerTypeToPayloadCallback(
            visualizationMessages.DetailsView.Open,
            this.onDetailsViewOpen,
        );
        this.interpreter.registerTypeToPayloadCallback(
            visualizationMessages.DetailsView.Select,
            this.onPivotChildSelected,
        );
        this.interpreter.registerTypeToPayloadCallback(
            visualizationMessages.DetailsView.PivotSelect,
            this.onDetailsViewPivotSelected,
        );
        this.interpreter.registerTypeToPayloadCallback(
            visualizationMessages.DetailsView.Close,
            this.onDetailsViewClosed,
        );
        this.interpreter.registerTypeToPayloadCallback(
            Messages.Assessment.AssessmentScanCompleted,
            this.onAssessmentScanCompleted,
        );
        this.interpreter.registerTypeToPayloadCallback(
            Messages.Assessment.StartOverTest,
            this.onStartOver,
        );
        this.interpreter.registerTypeToPayloadCallback(
            Messages.Assessment.CancelStartOver,
            this.onCancelStartOver,
        );
        this.interpreter.registerTypeToPayloadCallback(
            Messages.Assessment.StartOverAllAssessments,
            this.onStartOverAllAssessments,
        );
        this.interpreter.registerTypeToPayloadCallback(
            Messages.Assessment.CancelStartOverAllAssessments,
            this.onCancelStartOverAllAssessments,
        );
        this.interpreter.registerTypeToPayloadCallback(
            Messages.Assessment.EnableVisualHelper,
            this.onEnableVisualHelper,
        );
        this.interpreter.registerTypeToPayloadCallback(
            Messages.Assessment.DisableVisualHelperForTest,
            this.onDisableVisualHelpersForTest,
        );
        this.interpreter.registerTypeToPayloadCallback(
            Messages.Assessment.DisableVisualHelper,
            this.onDisableVisualHelper,
        );
        this.interpreter.registerTypeToPayloadCallback(
            Messages.Assessment.EnableVisualHelperWithoutScan,
            this.onEnableVisualHelperWithoutScan,
        );

        this.interpreter.registerTypeToPayloadCallback(
            Messages.Inspect.SetHoveredOverSelector,
            this.onSetHoveredOverSelector,
        );
    }

    private onEnableVisualHelperWithoutScan = (payload: ToggleActionPayload): void => {
        this.visualizationActions.enableVisualizationWithoutScan.invoke(payload);
    };

    private onEnableVisualHelper = (payload: ToggleActionPayload): void => {
        this.visualizationActions.enableVisualization.invoke(payload);
    };

    private onDisableVisualHelpersForTest = (payload: ToggleActionPayload): void => {
        this.visualizationActions.disableVisualization.invoke(payload.test);
    };

    private onDisableVisualHelper = (payload: ToggleActionPayload): void => {
        const eventName = TelemetryEvents.DISABLE_VISUAL_HELPER;
        this.telemetryEventHandler.publishTelemetry(eventName, payload);
        this.visualizationActions.disableVisualization.invoke(payload.test);
    };

    private onStartOver = (payload: ToggleActionPayload): void => {
        const eventName = TelemetryEvents.START_OVER_TEST;
        this.telemetryEventHandler.publishTelemetry(eventName, payload);
        this.visualizationActions.disableVisualization.invoke(payload.test);
    };

    private onCancelStartOver = (payload: BaseActionPayload): void => {
        const eventName = TelemetryEvents.CANCEL_START_OVER_TEST;
        this.telemetryEventHandler.publishTelemetry(eventName, payload);
    };

    private onStartOverAllAssessments = (payload: ToggleActionPayload): void => {
        const eventName = TelemetryEvents.START_OVER_ASSESSMENT;
        this.telemetryEventHandler.publishTelemetry(eventName, payload);
        this.visualizationActions.disableAssessmentVisualizations.invoke(null);
    };

    private onCancelStartOverAllAssessments = (payload: BaseActionPayload): void => {
        const eventName = TelemetryEvents.CANCEL_START_OVER_ASSESSMENT;
        this.telemetryEventHandler.publishTelemetry(eventName, payload);
    };

    private onDetailsViewClosed = (): void => {
        this.visualizationActions.disableAssessmentVisualizations.invoke(null);
    };

    private onAssessmentScanCompleted = async (
        payload: ScanCompletedPayload<any>,
        tabId: number,
    ): Promise<void> => {
        const eventName = TelemetryEvents.ASSESSMENT_SCAN_COMPLETED;
        this.telemetryEventHandler.publishTelemetry(eventName, payload);
        this.visualizationActions.scanCompleted.invoke(null);
        this.notificationCreator.createNotificationByVisualizationKey(
            payload.selectorMap,
            payload.key,
            payload.testType,
            payload.scanIncompleteWarnings,
        );
        await this.targetTabController.showTargetTab(tabId, payload.testType, payload.key);
    };

    private onTabbedElementAdded = (payload: AddTabbedElementPayload): void => {
        this.visualizationScanResultActions.addTabbedElement.invoke(payload);
    };

    private onRecordingCompleted = (payload: BaseActionPayload): void => {
        this.telemetryEventHandler.publishTelemetry(
            TelemetryEvents.TABSTOPS_RECORDING_COMPLETE,
            payload,
        );
    };

    private onRecordingTerminated = (payload: BaseActionPayload): void => {
        this.visualizationScanResultActions.disableTabStop.invoke(payload);
    };

    private onUpdateFocusedInstance = (payload: string[]): void => {
        this.visualizationActions.updateFocusedInstance.invoke(payload);
    };

    private onAdHocScanCompleted = async (
        payload: ScanCompletedPayload<any>,
        tabId: number,
    ): Promise<void> => {
        const telemetryEventName = TelemetryEvents.ADHOC_SCAN_COMPLETED;
        this.telemetryEventHandler.publishTelemetry(telemetryEventName, payload);
        this.visualizationScanResultActions.scanCompleted.invoke(payload);
        this.visualizationActions.scanCompleted.invoke(null);
        this.notificationCreator.createNotificationByVisualizationKey(
            payload.selectorMap,
            payload.key,
            payload.testType,
            payload.scanIncompleteWarnings,
        );
        await this.targetTabController.showTargetTab(tabId, payload.testType, payload.key);
    };

    private onScrollRequested = (): void => {
        this.visualizationActions.scrollRequested.invoke(null);
        this.cardSelectionActions.resetFocusedIdentifier.invoke(null);
    };

    private onDetailsViewOpen = async (
        payload: OnDetailsViewOpenPayload,
        tabId: number,
    ): Promise<void> => {
        if (this.shouldEnableToggleOnDetailsViewOpen(payload.detailsViewType)) {
            this.enableToggleOnDetailsViewOpen(payload.detailsViewType, tabId);
        }

        await this.onPivotChildSelected(payload, tabId);
    };

    private shouldEnableToggleOnDetailsViewOpen(visualizationType: VisualizationType): boolean {
        return (
            visualizationType != null &&
            visualizationType !== VisualizationType.TabStops &&
            this.visualizationConfigurationFactory.getConfiguration(visualizationType).testMode !==
                TestMode.Assessments
        );
    }

    private enableToggleOnDetailsViewOpen(test: VisualizationType, tabId: number): void {
        const payload: VisualizationTogglePayload =
            this.createVisualizationTogglePayloadWithNullTelemetry(test);
        this.onVisualizationToggle(payload);
    }

    private createVisualizationTogglePayloadWithNullTelemetry(
        test: VisualizationType,
    ): VisualizationTogglePayload {
        return {
            test,
            enabled: true,
            telemetry: null,
        };
    }

    private onPivotChildSelected = async (
        payload: OnDetailsViewOpenPayload,
        tabId: number,
    ): Promise<void> => {
        this.sidePanelActions.closeSidePanel.invoke('PreviewFeatures');
        this.visualizationActions.updateSelectedPivotChild.invoke(payload);
        await this.detailsViewController
            .showDetailsView(tabId)
            .catch(e => this.logger.error(e.message));
        this.telemetryEventHandler.publishTelemetry(TelemetryEvents.PIVOT_CHILD_SELECTED, payload);
    };

    private onDetailsViewPivotSelected = (payload: OnDetailsViewPivotSelected): void => {
        this.visualizationActions.updateSelectedPivot.invoke(payload);
        this.telemetryEventHandler.publishTelemetry(
            TelemetryEvents.DETAILS_VIEW_PIVOT_ACTIVATED,
            payload,
        );
    };

    private onVisualizationToggle = (payload: VisualizationTogglePayload): void => {
        const telemetryEvent = this.adHocTestTypeToTelemetryEvent[payload.test];
        this.telemetryEventHandler.publishTelemetry(telemetryEvent, payload);
        this.unifiedScanResultActions.startScan.invoke(null);

        if (payload.enabled) {
            this.visualizationActions.enableVisualization.invoke(payload);
        } else {
            this.visualizationActions.disableVisualization.invoke(payload.test);
        }
    };

    private onRescanVisualization = (payload: RescanVisualizationPayload) => {
        this.visualizationActions.disableVisualization.invoke(payload.test);
        this.visualizationActions.enableVisualization.invoke(payload);
        this.unifiedScanResultActions.startScan.invoke(null);
        this.telemetryEventHandler.publishTelemetry(TelemetryEvents.RESCAN_VISUALIZATION, payload);
    };

    private getVisualizationToggleCurrentState = (): void => {
        this.visualizationActions.getCurrentState.invoke(null);
    };

    private getScanResultsCurrentState = (): void => {
        this.visualizationScanResultActions.getCurrentState.invoke(null);
    };

    private onSetHoveredOverSelector = (payload: string[]): void => {
        this.inspectActions.setHoveredOverSelector.invoke(payload);
    };
}
