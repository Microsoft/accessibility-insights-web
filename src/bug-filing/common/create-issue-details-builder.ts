// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { EnvironmentInfo } from '../../common/environment-info-provider';
import { CreateIssueDetailsTextData } from '../../common/types/create-issue-details-text-data';
import { title } from '../../content/strings/application';
import { IssueDetailsBuilder } from './issue-details-builder';
import { MarkupFactory } from './markup-factory';

export const createIssueDetailsBuilder = (markup: MarkupFactory): IssueDetailsBuilder => {
    const getter = (environmentInfo: EnvironmentInfo, data: CreateIssueDetailsTextData): string => {
        const result = data.ruleResult;

        const text = [
            `${markup.bold('Issue')}: ${markup.snippet(result.help)} (${markup.link(result.helpUrl, result.ruleId)})`,
            markup.sectionSeparator(),
            `${markup.bold('Target application')}: ${markup.link(data.pageUrl, data.pageTitle)}`,
            markup.sectionSeparator(),
            `${markup.bold('Element path')}: ${data.ruleResult.selector}`,
            markup.sectionSeparator(),
            `${markup.bold('Snippet')}:`,
            markup.snippet(result.snippet),
            markup.sectionSeparator(),
            `${markup.bold('How to fix')}:`,
            markup.newLine(),
            markup.howToFixSection(result.failureSummary),
            markup.sectionSeparator(),
            `${markup.bold('Environment')}:`,
            environmentInfo.browserSpec,
            markup.sectionSeparator(),
            `This accessibility issue was found using ${title} ` +
                `${environmentInfo.extensionVersion} (axe-core ${environmentInfo.axeCoreVersion}), ` +
                'a tool that helps find and fix accessibility issues. Get more information & download ' +
                `this tool at ${markup.link('http://aka.ms/AccessibilityInsights')}.`,
        ].join('\n');

        return text;
    };

    return getter;
};
