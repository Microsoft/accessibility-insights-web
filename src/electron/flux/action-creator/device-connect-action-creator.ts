// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { TelemetryEventHandler } from 'background/telemetry/telemetry-event-handler';
import { VALIDATE_PORT } from '../../common/electron-telemetry-events';

export class DeviceConnectActionCreator {
    constructor(private readonly telemetryEventHandler: TelemetryEventHandler) {}

    public validatePort(port: number): void {
        this.telemetryEventHandler.publishTelemetry(VALIDATE_PORT, { telemetry: { port } });
    }
}
