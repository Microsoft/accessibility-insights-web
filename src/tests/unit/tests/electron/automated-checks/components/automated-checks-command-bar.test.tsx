// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import { Button } from 'office-ui-fabric-react/lib/Button';
import * as React from 'react';
import { Mock, MockBehavior, Times } from 'typemoq';

import {
    AutomatedChecksCommandBar,
    AutomatedChecksCommandBarProps,
} from 'electron/automated-checks/components/automated-checks-command-bar';
import { DeviceConnectActionCreator } from 'electron/flux/action-creator/device-connect-action-creator';
import { EventStubFactory } from 'tests/unit/common/event-stub-factory';

describe('AutomatedChecksCommandBar', () => {
    test('render', () => {
        const props: AutomatedChecksCommandBarProps = { deps: { deviceConnectActionCreator: null } };
        const rendered = shallow(<AutomatedChecksCommandBar {...props} />);

        expect(rendered.getElement()).toMatchSnapshot();
    });

    test('rescan click', () => {
        const eventStub = new EventStubFactory().createMouseClickEvent() as React.MouseEvent<Button>;

        const deviceConnectActionCreatorMock = Mock.ofType<DeviceConnectActionCreator>(undefined, MockBehavior.Strict);
        deviceConnectActionCreatorMock.setup(creator => creator.resetConnection()).verifiable(Times.once());

        const props = {
            deps: {
                deviceConnectActionCreator: deviceConnectActionCreatorMock.object,
            },
        } as AutomatedChecksCommandBarProps;

        const rendered = shallow(<AutomatedChecksCommandBar {...props} />);
        const button = rendered.find('.button-rescan');

        button.simulate('click', eventStub);

        deviceConnectActionCreatorMock.verifyAll();
    });
});
