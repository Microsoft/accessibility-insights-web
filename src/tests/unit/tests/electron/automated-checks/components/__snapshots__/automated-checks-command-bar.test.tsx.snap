// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutomatedChecksCommandBar render 1`] = `
<div
  className="automatedChecksCommandBar"
>
  <CustomizedActionButton
    className="button-rescan"
    iconProps={
      Object {
        "iconName": "Refresh",
      }
    }
    onClick={[Function]}
    text="Rescan"
  />
</div>
`;
